source/compiler-core/slang-dxc-compiler.cpp

```diff
@@ -572,6 +573,20 @@ SlangResult DXCDownstreamCompiler::convert(IArtifact* from, const ArtifactDesc&
     return SLANG_OK;
 }
 
+SlangResult DXCDownstreamCompiler::getVersionString(slang::IBlob** outVersionString)
+{
+    if (m_desc.hasVersion())
+    {
+        auto version = m_desc.getVersionValue();
```

Comment:
The version field in `m_desc` is never filled in. 
The right way to get version is to create an instance of dxc by calling `m_createInstance`, then query the `IDxcVersionInfo` interface on the dxc instance, this interface provides a `GetVersion` method that returns the verison number.

A better way is just to hash the dxcompiler.dll and use that digest as the version string. This is more robust against out-of-band builds of dxc.